File: api/src/main/java/io/matchmore/sdk/MatchesApi.java
package io.matchmore.sdk;

import io.matchmore.CollectionFormats.*;



import retrofit2.Call;
import retrofit2.http.*;

import okhttp3.RequestBody;

import io.matchmore.sdk.models.APIError;
import io.matchmore.sdk.models.Matches;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;


public interface MatchesApi {
  /**
   * Get matches for the device
   * 
   * @param deviceId The id (UUID) of the device. (required)
   * @return Call&lt;Matches&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @GET("devices/{deviceId}/matches")
  Call<Matches> getMatches(
    @retrofit2.http.Path("deviceId") String deviceId
  );

}

File: sdk/src/main/java/io/matchmore/sdk/api/MatchesApi.kt
package io.matchmore.sdk.api


import io.matchmore.sdk.api.models.Matches
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.Path


interface MatchesApi {
    /**
     * Get matches for the device
     *
     * @param deviceId The id (UUID) of the device. (required)
     * @return Call&lt;Matches&gt;
     */
    @Headers("Content-Type:application/json")
    @GET("devices/{deviceId}/matches")
    fun getMatches(
            @Path("deviceId") deviceId: String
    ): Call<Matches>

}
